src/utilities/dialogs/Overlay.tsx

```diff
@@ -0,0 +1,79 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useMemoizedId } from '@hooks/useMemoizedId';
+import { Popper } from '@components/Popper';
+import React, { Children, cloneElement, isValidElement } from 'react';
+import { Trigger } from '@utilities/dialogs/Trigger';
+import { Content } from '@utilities/dialogs/Content';
+import { OVERLAY_CONTAINER_CLASSES } from '@utilities/overlayStyle';
+import { BaseDialogProps, Modality, OverlayProps } from '../../types/dialog';
+import { merge } from '@utilities/merge';
+import { useFocusTrap } from '@hooks/useFocusTrap';
+import { useClickOutside } from '@hooks/useClickOutside';
+
+export const Overlay = ({
+    open,
+    placement,
+    offset,
+    flip,
+    enablePortal,
+    children,
+    role = 'region',
+    'data-test-id': dataTestId = 'fondue-overlay',
+    maxWidth = 'auto',
+    maxHeight = 'auto',
+    handleClose,
+    modality,
+}: OverlayProps & BaseDialogProps) => {
+    const id = useMemoizedId();
+    const ref = useFocusTrap(modality);
+    const handleClosingInteraction = () => {
+        if (open && modality !== Modality.BlockingModal) {
+            if (handleClose) {
+                handleClose();
+            }
+        }
+        return;
+    };
+    useClickOutside(ref.current, handleClosingInteraction);
+    return (
+        <Popper open={open} placement={placement} offset={offset} flip={flip} enablePortal={enablePortal}>
+            {Children.map(children, (child) => {
+                if (isValidElement(child) && typeof child.type === 'function') {
+                    const { name } = child.type;
+
+                    if (name === Trigger.name) {
+                        return (
+                            <Popper.Reference>
+                                <div className="tw-w-fit" id={id}>
```

Comment:
is this necessary? I think this div could cause issues in the future that we've had in the past where children components can no longer determine their width based on the parent component